### PR TITLE
cmd/soroban-rpc: Add txsub metrics

### PR DESCRIPTION
**Pull request by tamirms**
### What

Add metrics to:

-  count transaction submission counts / duration broken down by status.
- count number of operations per envelope broken down by tx sub status.

### Why

https://github.com/stellar/soroban-tools/issues/496

### Known limitations

[N/A]

### CODE REVIEW

File: cmd/soroban-rpc/internal/daemon/metrics.go [hunk 1]
```diff
@@ -40,3 +47,56 @@ func (d *Daemon) MetricsRegistry() *prometheus.Registry {
...
 }
+
+type CoreClientWithMetrics struct {
+	stellarcore.Client
+	submitMetric  *prometheus.SummaryVec
+	opCountMetric *prometheus.SummaryVec
+}
+
+func newCoreClientWithMetrics(client stellarcore.Client, registry *prometheus.Registry) *CoreClientWithMetrics {
+	submitMetric := prometheus.NewSummaryVec(prometheus.SummaryOpts{
```
Comment (paulbellamy) [inline cmd/soroban-rpc/internal/daemon/metrics.go:58]: Should we aim to match the metrics we emit from horizon txsub? wdyt?
https://github.com/stellar/go/blob/master/services/horizon/internal/txsub/system.go#L410

Comment (paulbellamy) [reply cmd/soroban-rpc/internal/daemon/metrics.go:58]: No, the concern is that it seems like horizon tracks these a bit more in-depth. open/buffered/failed, etc. I like using the response status, but we don't currently get visibility into the actual txsub queueing. But I guess we don't have any queueing here, so txsub is a bit simpler... 🤔

Comment (paulbellamy) [reply cmd/soroban-rpc/internal/daemon/metrics.go:58]: (And that it would be nice to match naming and metrics conventions for consistency)

**Review by paulbellamy — APPROVED** [review]

**Review by tsachiherman — APPROVED** [review]
Looks great. I'm not sure if these metrics are overkill as Paul mentioned.

**Pull request merged**